From stern@rowland.harvard.edu Tue Oct 10 08:56:16 2006
Date: Tue, 10 Oct 2006 11:55:47 -0400 (EDT)
From: Alan Stern <stern@rowland.harvard.edu>
To: Greg KH <greg@kroah.com>
cc: USB development list <linux-usb-devel@lists.sourceforge.net>
Subject: usbcore: fix refcount bug in endpoint removal
Message-ID: <Pine.LNX.4.44L0.0610101153070.6128-100000@iolanthe.rowland.org>

This patch (as799) fixes a nasty refcount error in the USB endpoint class.

Signed-off-by: Alan Stern <stern@rowland.harvard.edu>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/core/endpoint.c |    4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

--- gregkh-2.6.orig/drivers/usb/core/endpoint.c
+++ gregkh-2.6/drivers/usb/core/endpoint.c
@@ -282,8 +282,6 @@ void usb_remove_ep_files(struct usb_host
 		sysfs_remove_group(&endpoint->ep_dev->dev.kobj, &ep_dev_attr_grp);
 		device_unregister(&endpoint->ep_dev->dev);
 		endpoint->ep_dev = NULL;
+		destroy_endpoint_class();
 	}
-	destroy_endpoint_class();
 }
-
-
